sql/rpl_rli.h

```diff
@@ -654,8 +654,7 @@ struct start_alter_info
   /*
     ALTER id is defined as a pair of GTID's seq_no and domain_id.
   */
-  decltype(rpl_gtid::seq_no) sa_seq_no; // key for searching (SA's id)
-  uint32 domain_id;
+  rpl_gtid gtid; // rpl_gtid::seq_no is the key for searching (SA's id)
   bool   direct_commit_alter; // when true CA thread executes the whole query
```

Comment:
This (as opposed to `rpl_gtid *gtid`) will put the GTID in the memory next to sibling members like `direct_commit_alter` (at least in C). This also matches `sa_seq_no` and `domain_id`.

I’m not sure about this “key” designation in the comments, though – I didn’t encounter any “keying” while migrating affected code.